newIDE/app/src/Profile/Subscription/SubscriptionDialog.js

```diff
@@ -192,85 +195,87 @@ export default function SubscriptionDialog({
-        try {
-          await changeUserSubscription(getAuthorizationHeader, profile.id, {
-            planId: plan.planId,
-          });
-          await authenticatedUser.onRefreshSubscription();
-          showAlert({
-            title: t`Subscription updated`,
-            message: t`Congratulations, your new subscription is now active! You can now use the services unlocked with this plan.`,
-          });
-        } catch (rawError) {
-          showErrorBox({
-            message: i18n._(
-              t`Your subscription could not be updated. Please try again later!`
-            ),
-            rawError,
-            errorId: 'subscription-update-error',
-          });
-        } finally {
-          setIsChangingSubscription(false);
-        }
-      } else {
-        // Changing the existing subscription by cancelling first.
-        setIsChangingSubscription(true);
-        try {
-          await changeUserSubscription(getAuthorizationHeader, profile.id, {
-            planId: '',
-          });
-          await authenticatedUser.onRefreshSubscription();
-        } catch (rawError) {
-          showErrorBox({
-            message: i18n._(
-              t`Your subscription could not be cancelled. Please try again later!`
-            ),
-            rawError,
-            errorId: 'subscription-update-error',
-          });
-        } finally {
-          setIsChangingSubscription(false);
-        }
+    if (!shouldSkipAlert) {
+      const answer = await showConfirmation(confirmDialogTexts);
+      if (!answer) return;
+    }
 
-        // Then redirect as if a new subscription is being chosen.
-        setSubscriptionPendingDialogOpen(true);
-        Window.openExternalURL(
-          getRedirectToCheckoutUrl({
-            planId,
-            userId: profile.id,
-            userEmail: profile.email,
-          })
-        );
+    if (!needToCancelSubscription) {
+      // Changing the existing subscription without asking for payment details again.
+      setIsChangingSubscription(true);
+      try {
+        await changeUserSubscription(getAuthorizationHeader, profile.id, {
+          planId,
+        });
+        await authenticatedUser.onRefreshSubscription();
+        showAlert({
+          title: t`Subscription updated`,
+          message: t`Congratulations, your new subscription is now active! You can now use the services unlocked with this plan.`,
+        });
+      } catch (rawError) {
+        showErrorBox({
+          message: i18n._(
+            t`Your subscription could not be updated. Please try again later!`
+          ),
+          rawError,
+          errorId: 'subscription-update-error',
+        });
+      } finally {
+        setIsChangingSubscription(false);
       }
+    } else {
+      // Changing the existing subscription by cancelling first.
+      setIsChangingSubscription(true);
+      try {
+        await changeUserSubscription(getAuthorizationHeader, profile.id, {
+          planId: null,
```

Comment:
is null on purpose? Or should it stay as `''`?

Reply:
Yes, null was used elsewhere and it changes nothing to the logic in `changeUserSubscription`